src/pages/basic_data/institution_admin/edit_user/UserFormDialog.tsx

```diff
@@ -35,44 +38,18 @@ interface UserFormDialogProps extends Pick<DialogProps, 'open'> {
 
 export const UserFormDialog = ({ open, onClose, existingUser, existingPerson }: UserFormDialogProps) => {
   const { t } = useTranslation();
-  const user = useSelector((store: RootState) => store.user);
-  const topOrgCristinId = user?.topOrgCristinId;
-  const customerId = user?.customerId ?? '';
-
-  const personId = typeof existingPerson === 'string' ? existingPerson : existingPerson.id;
-  const existingPersonObject = typeof existingPerson === 'object' ? existingPerson : undefined;
-
-  const personQuery = useFetchProtectedPerson(personId, { enabled: open && !existingPersonObject });
-  const person = existingPersonObject ?? personQuery.data;
-  const personEmployments = person?.employments ?? [];
-
-  const topOrgCristinIdentifier = topOrgCristinId ? getIdentifierFromId(topOrgCristinId) : '';
-  const internalEmployments: Employment[] = [];
-  const externalEmployments: Employment[] = [];
-  const targetOrganizationIdStart = `${topOrgCristinIdentifier.split('.')[0]}.`;
-
-  personEmployments.forEach((employment) => {
-    const organizationIdentifier = employment.organization.split('/').pop();
-    if (organizationIdentifier?.startsWith(targetOrganizationIdStart)) {
-      internalEmployments.push(employment);
-    } else {
-      externalEmployments.push(employment);
-    }
-  });
-
+  const { topOrgCristinId, customerId } = useLoggedInUser();
+  const { person, personQuery } = useProtectedPerson(existingPerson, open);
   const username = getUsername(person, topOrgCristinId);
-
-  const institutionUserQuery = useFetchUserQuery(username, {
-    enabled: open && !existingUser,
+  const { institutionUser, institutionUserQuery } = useInstitutionUser({
+    existingUser,
+    username,
+    enabled: open, // Only fetch when dialog is open
```

Comment:
Tenker at det er bedre å endre variabelnavn enn å måtte ha en kommentar her. Mener forsåvidt også at det man skjønner hva som skjer her utifra context.
```suggestion
    enabled: open, 
```